Libraries/CustomComponents/ListView/ListView.js

```diff
@@ -484,9 +484,8 @@ var ListView = React.createClass({
   },
 
   _getDistanceFromEnd: function(scrollProperties) {
-    return scrollProperties.contentLength -
-      scrollProperties.visibleLength -
-      scrollProperties.offset;
+    var maxHeight = Math.max(scrollProperties.contentLength, scrollProperties.visibleLength);
```

Comment:
Separate PR sounds good since it's easier to go through the commit logs and bisect if there's an issue. Could you rename `maxHeight` to `maxLength` since the ScrollView could be horizontal, and wrap the code to fit 80 cols?